test/integration/deployment/util.go

```diff
@@ -351,3 +349,26 @@ func (d *deploymentTester) waitForDeploymentUpdatedReplicasLTE(minUpdatedReplica
 func (d *deploymentTester) waitForDeploymentWithCondition(reason string, condType v1beta1.DeploymentConditionType) error {
 	return testutil.WaitForDeploymentWithCondition(d.c, d.deployment.Namespace, d.deployment.Name, reason, condType, d.t.Logf, pollInterval, pollTimeout)
 }
+
+func (d *deploymentTester) listPods(selector labels.Selector) ([]v1.Pod, error) {
```

Comment:
no need to pass selector, because you can get it from tester

Reply:
@janetkuo How should we get `selector` from tester? I prefer not to call `metav1.LabelSelectorAsSelector(d.deployment.Spec.Selector)` while polling.

Reply:
Since it lists pods from the deployment stored in the tester it's easier and safer to get selector from tester -- it prevents the caller to pass in non-matching selector.